tripal/tripal
# Update or add all remaining linking fields from issue #1381

# Tripal 4 Core Dev Task

### Closes #1381
### Closes #1387

### Tripal Version: 4.x

## Description
This pull request will upgrade or add all the remaining fields mentioned in https://github.com/tripal/tripal/issues/1381#issuecomment-1823789804

These fields will include "linker capabilities" so that the field can be used directly on a base table, or connected through a linking table.

~~This is dependent on PR #1697~~ (now merged) which adds Analysis and updates Contact and the base class.
~~This is dependent on PR #1710~~ (now merged) which provides the ChadoDbxrefDefault field.


~~This will include a fix for issue #1703, which can't be a separate PR because we need the fields first to test the fix! These are the changes in `tripal_chado/src/Plugin/TripalStorage/ChadoStorage.php`~~ This was reverted and #1729 fixed this instead.

This is a draft until I finish the yaml additions

## Testing?
1. Build a docker on this branch
```
port=8080
git clone https://github.com/tripal/tripal.git testing-1381
cd testing-1381
git checkout tv4g1-issue1381-chado_linker_remaining_fields
docker build --tag=tripaldocker:testing-1381 --build-arg drupalversion="10.1.x-dev" --build-arg chadoschema='chado' --file tripaldocker/Dockerfile-php8.2-pgsql13 ./
docker run --publish=${port}:80 --name=1381 -tid --volume=$(pwd):/var/www/drupal/web/modules/contrib/tripal tripalproject/tripaldocker:latest
docker exec testing-1381 service postgresql start
```
2. Create some content for testing: Organism, Contact, Publication. Make sure to put unique content in each of the publication fields to test them all. For ELocation try something with a `https:` prefix to make sure it displays as a link
![1381-2](https://github.com/tripal/tripal/assets/8419404/200afc19-5013-4513-bc4e-da1e0afd2534)

3. Create a Study, and specify both contact and publication when you create it. This tests ~~the fix for issue #1703~~ the current version of chado storage - there should  be no error.
4. Import all the other content types: Page Structure ->  + Import Type Collection -> Check all boxes and click "Import"
`drush trp-run-jobs --username=drupaladmin --root=/var/www/drupal/web`
There should of course be no errors.
5. Create an Array Design, then create an Assay using that Array Design
6. ...
